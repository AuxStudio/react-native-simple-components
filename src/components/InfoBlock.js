import React from "react";
import { View, ScrollView, Text, StyleSheet } from "react-native";

import styleConstants from "../assets/styleConstants";

import Touchable from "./Touchable";

const styles = StyleSheet.create({
    infoContainer: {
        paddingRight: 96,
        paddingLeft: 16,
        paddingBottom: 16,
    },
    infoTextTitle: {
        fontSize: styleConstants.largeFont,
        color: styleConstants.white,
    },
    descriptionWrapper: {},
    descriptionContainer: {},
    infoTextDescription: {
        marginTop: 8,
        fontSize: styleConstants.regularFont,
        color: styleConstants.white,
    },
});

export default (InfoBlock = props => {
    /* 
        PROPTYPES

        title: PropTypes.string,
        subtitle: PropTypes.string,
        titleColor: PropTypes.string,
        subtitleColor: PropTypes.string,
        fullWidth: PropTypes.bool,
        flex: PropTypes.bool,
    */

    const fullWidthStyles = props.fullWidth && {
        paddingRight: 16,
    };

    const flexStyles = props.flex && {
        flex: 1,
    };

    const fullWidthTitleStyles = props.fullWidth && {
        marginRight: 16,
    };

    const subtitle = props.flex ? (
        <ScrollView
            style={styles.descriptionWrapper}
            contentContainerStyles={styles.descriptionContainer}>
            <Text
                style={[
                    styles.infoTextDescription,
                    { color: props.subtitleColor },
                    styleConstants.primaryFont,
                ]}>
                {props.subtitle}
            </Text>
        </ScrollView>
    ) : (
        <Text
            style={[
                styles.infoTextDescription,
                { color: props.subtitleColor },
                styleConstants.primaryFont,
            ]}>
            {props.subtitle}
        </Text>
    );

    return (
        <View style={[styles.infoContainer, fullWidthStyles, flexStyles]}>
            <Text
                style={[
                    styles.infoTextTitle,
                    { color: props.titleColor },
                    styleConstants.primaryFont,
                    fullWidthTitleStyles,
                ]}>
                {props.title}
            </Text>
            {subtitle}
        </View>
    );
});
