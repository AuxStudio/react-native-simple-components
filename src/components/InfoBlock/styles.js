import { StyleSheet } from 'react-native';

import styleConstants from '../../styleConstants';

const styles = StyleSheet.create({
  container: {
    alignSelf: 'stretch',
  },
  titleText: {
    fontSize: styleConstants.fonts.sizes.large,
    color: styleConstants.colors.primaryText,
    marginBottom: styleConstants.dimensions.margin.small,
  },
  descriptionText: {
    fontSize: styleConstants.fonts.sizes.regular,
    color: styleConstants.colors.secondaryText,
  },
});

export default styles;
