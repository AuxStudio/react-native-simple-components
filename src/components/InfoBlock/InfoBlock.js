import React from 'react';
import { View, ScrollView, Text } from 'react-native';

import styles from './styles';
import styleConstants from '../assets/styleConstants';

import Touchable from './Touchable';

export default (InfoBlock = (props) => {
  /* 
        PROPTYPES

        title: PropTypes.string,
        description: PropTypes.string,

        // titleTextStyle: PropTypes.node,
        // descriptionTextStyle: PropTypes.node,
    */

  return (
    <View style={styles.infoContainer}>
      <Text style={[styles.infoTextTitle, props.titleTextStyle]}>{props.title}</Text>
      <Text style={[styles.infoTextDescription, props.descriptionTextStyle]}>
        {props.description}
      </Text>
    </View>
  );
});
