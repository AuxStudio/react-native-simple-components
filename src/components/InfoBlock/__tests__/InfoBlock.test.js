/* eslint-disable import/first */
import React from 'react';
import renderer from 'react-test-renderer';

import InfoBlock from '../';

it('renders a InfoBlock', () => {
  expect(
    renderer.create(
      <InfoBlock
        title="Title"
        description="Description"
        titleTextStyle={{ color: 'red' }}
        descriptionTextStyle={{ color: 'red' }}
        style={{ backgroundColor: 'blue' }}
      />,
    ),
  ).toMatchSnapshot();
});

it('renders a InfoBlock with no props', () => {
  expect(renderer.create(<InfoBlock />)).toMatchSnapshot();
});
/* eslint-enable */
