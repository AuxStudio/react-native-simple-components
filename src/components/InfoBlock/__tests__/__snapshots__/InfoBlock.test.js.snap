// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a InfoBlock 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "stretch",
      },
      Object {
        "backgroundColor": "blue",
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#212121",
          "fontSize": 24,
          "marginBottom": 8,
        },
        Object {
          "color": "red",
        },
      ]
    }
  >
    Title
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#757575",
          "fontSize": 17,
        },
        Object {
          "color": "red",
        },
      ]
    }
  >
    Description
  </Text>
</View>
`;

exports[`renders a InfoBlock with no props 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "stretch",
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#212121",
          "fontSize": 24,
          "marginBottom": 8,
        },
        undefined,
      ]
    }
  >
    Title
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#757575",
          "fontSize": 17,
        },
        undefined,
      ]
    }
  >
    Description
  </Text>
</View>
`;
